miscelanea/005-deploy-in-one-pod/freeipa.yaml.envsubst

```diff
@@ -0,0 +1,448 @@
+# Proof of Concept that launch ipa-server-install in an initContainer,
+# preparing the whole data volume.
+#
+# Quick Start:
+#   make app-delete app-deploy get-info
+#
+# References:
+#   https://systemd.io/CONTAINER_INTERFACE/
+#   https://stackoverflow.com/questions/52940774/kubernetes-how-to-check-current-domain-set-by-cluster-domain-from-pod
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: freeipa
+  labels:
+    app: freeipa
+data:
+  DEBUG_TRACE: "2"
+  IPA_SERVER_HOSTNAME: "${APP}.apps.${CLUSTER_DOMAIN}"
+  IPA_SERVER_IP: "10.0.135.189"
+  # TODO Move PASSWORD into a Secret object
+  PASSWORD: "Secret123"
+  container: "podman"
+  container_uuid: ""
+  UID_BASE: ""
+  GID_BASE: ""
+  # yamllint disable rule:line-length
+  init-uid-gid-base.sh: |
+    #!/bin/bash
+    function yield {
+      echo "$*" >&2
+    }
+    function verbose {
+      yield "$*"
+      "$@"
+    }
+    echo "Script: $0"
+    TOKEN="$(</var/run/secrets/kubernetes.io/serviceaccount/token)"
+    CA_CERT="/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+    echo "NAMESPACE=$DOLLAR{NAMESPACE}"
+    oc login --token ""  \
+             --certificate-authority="" \
+             https://kubernetes.default.svc:443
+    if [ "$DOLLAR{UID_BASE}" == "" ]; then
+      UID_BASE="$( oc get namespace/$DOLLAR{NAMESPACE} \
+                   -o 'jsonpath={.metadata.annotations.openshift\.io/sa\.scc\.uid-range}' )"
+      UID_BASE="$DOLLAR{UID_BASE%/*}"
+      oc patch configmap freeipa --patch "{\"data\":{\"UID_BASE\": \"$DOLLAR{UID_BASE}\"}}"
+    fi
+    if [ "$DOLLAR{GID_BASE}" == "" ]; then
+      GID_BASE="$( oc get namespace/$DOLLAR{NAMESPACE} \
+                   -o 'jsonpath={.metadata.annotations.openshift\.io/sa\.scc\.supplemental-groups}' )"
+      GID_BASE="$DOLLAR{GID_BASE%/*}"
+      oc patch configmap freeipa \
+         --patch "{\"data\":{\"GID_BASE\":\"$DOLLAR{GID_BASE}\"}}"
+    fi
+    UID_BASE="$( oc get cm/freeipa -o 'jsonpath={.data.UID_BASE}' )"
+    GID_BASE="$( oc get cm/freeipa -o 'jsonpath={.data.GID_BASE}' )"
```

Comment:
These variables get set in the configmap, and are based on the namespace uid and gid ranges.

This is instructive, to see how it is done!  But it seems that nothing actually uses these.  And it is expected (hoped?) that userns support will dissolve any need to worry about what uids/gids are available to run as.

Reply:
Internal ticket: https://issues.redhat.com/browse/IDMOCP-184 . Resolving comment.